charts/simple-app/templates/deployment.yaml

```diff
@@ -253,6 +259,7 @@ spec:
                 name: {{ tpl (required ".name key must be set on secretRef" .name) $ }}
             {{- end }}
             {{- end }}
+            {{- end }}
```

Comment:
I feel like this extra `end` is wrong... you've added a new `if...end` statement, but its all self contained... am I wrong?

Reply:
This closes the `{{- with .Values.envFrom}}` on :251.